packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -190,3 +206,116 @@ export function getNetworkName(chainId: number) {
       return 'Unknown'
   }
 }
+
+export function getNetworkLogo(chainId: number) {
+  switch (chainId) {
+    // L1 networks
+    case ChainId.Mainnet:
+      return EthereumLogo
+
+    case ChainId.Rinkeby:
+      return EthereumLogo
+
+    case ChainId.Goerli:
+      return EthereumLogo
+
+    // L2 networks
+    case ChainId.ArbitrumOne:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumNova:
+      return ArbitrumNovaLogo
+
+    case ChainId.ArbitrumRinkeby:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumGoerli:
+      return ArbitrumOneLogo
+
+    default:
+      return EthereumLogo
+  }
+}
+
+type ExtendedWeb3Provider = Web3Provider & {
+  isMetaMask?: boolean
+  isImToken?: boolean
+}
+
+export const getSupportedNetworks = (chainId?: number) => {
+  // The list which will be available for network selection in navbar-dropdowns and unsupported-network-content
+  // If there is a chainId selected, it detects if it's a testnet and shows only those options
+
+  const mainNetworks = [
+    ChainId.ArbitrumOne,
+    ChainId.ArbitrumNova,
+    ChainId.Mainnet
+  ]
+  const testNetworks = [ChainId.ArbitrumGoerli, ChainId.Goerli]
+  const isTestnet = chainId ? isNetwork(chainId).isTestnet : false
+  return isTestnet ? testNetworks : mainNetworks
+}
+
+export type SwitchChainProps = {
+  chainId: number
+  provider: ExtendedWeb3Provider
+  onSuccess?: () => void
+  onError?: (err?: Error) => void
+  onSwitchChainNotSupported?: () => void
+}
+
+const isSwitchChainSupported = (provider: ExtendedWeb3Provider) => {
+  return provider?.provider?.isMetaMask || provider?.isImToken
```

Comment:
if we're checking for the first `provider` object potentially being `undefined`, we should add `undefined` to the arg type for this method and the `switchChain` method, and remove the `!` when `provider` is passed in (where it's called at @chrstph-dvx's comment)